kernel/src/engine/default/parquet.rs

```diff
@@ -245,6 +245,38 @@ impl<E: TaskExecutor> ParquetHandler for DefaultParquetHandler<E> {
             self.readahead,
         )
     }
+
+    fn read_parquet_schema(&self, file: &FileMeta) -> DeltaResult<SchemaRef> {
+        let store = self.store.clone();
+        let location = file.location.clone();
+        let file_size = file.size;
+
+        self.task_executor.block_on(async move {
+            let arrow_schema = if location.is_presigned() {
+                let client = reqwest::Client::new();
+                let response =
+                    client.get(location.as_str()).send().await.map_err(|e| {
+                        Error::generic(format!("Failed to fetch presigned URL: {}", e))
+                    })?;
+                let bytes = response
+                    .bytes()
+                    .await
+                    .map_err(|e| Error::generic(format!("Failed to read response bytes: {}", e)))?;
+                let metadata = ArrowReaderMetadata::load(&bytes, Default::default())?;
+                metadata.schema().clone()
```

Comment:
Let's just return the metadata out of the if/else instead of the schema, then you don't have to clone it. It's an arc clone but it's still good practice. So:

```rust
let metadata = if location.is_presigned() {
  ...
  ArrowReaderMetadata::load(&bytes, Default::default())?
} else {
  ...
  ArrowReaderMetadata::load_async(&mut reader, Default::default()).await?
};
StructType::try_from_arrow(metadata.schema())
    .map(Arc::new)
    .map_err(Error::Arrow)
```